include/dr/detail/index.hpp

```diff
@@ -9,7 +9,7 @@
 #include <limits>
 #include <tuple>
 
-namespace dr::shp {
+namespace dr {
```

Comment:
I think we have convention all stuff in "dr/detail" should be in namespace `dr::_detail`, not `dr`. I see some classes in dr/detail obey the rule and some don't, but I would use it for new classes.

@rscohn2 , please correct me if I'm wrong

Reply:
dr::index will be used as key type for accessing matrix elements - so it cannot be in __details, because users of matrix classes will use it directly.
Still, maybe we should provide dr::shp::index and dr::mhp::index as synonyms to dr::index, similar to what we do in views.